libs/dal/src/repositories/subscriber-preference/subscriber-preference.schema.ts

```diff
@@ -55,6 +52,16 @@ const subscriberPreferenceSchema = new Schema<SubscriberPreferenceDBModel>(
   schemaOptions
 );
 
+subscriberPreferenceSchema.index({
+  _subscriberId: 1,
+  _templateId: 1,
+});
+
+subscriberPreferenceSchema.index({
+  _subscriberId: 1,
+  level: 1,
```

Comment:
I added this here since we are doing a global check for each trigger (for both global and template specific). Index size will be quite small, due to low cardinality. But I would keep this one here just for the small boost in query time.